fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/MaterializedViewRewriter.java

```diff
@@ -866,6 +866,12 @@ protected OptExpression queryBasedRewrite(RewriteContext rewriteContext, ScalarO
                 Utils.compoundAnd(
                         rewriteContext.getQueryPredicateSplit().toScalarOperator(),
                         CompoundPredicateOperator.not(compensationPredicates)));
+        List<ScalarOperator> predicates = Utils.extractConjuncts(queryCompensationPredicate);
+        predicates.removeAll(mvRewriteContext.getOnPredicates());
```

Comment:
why remove

Reply:
because onPredicate is kept in Join, not necessary to add it again.

Reply:
a bit wried, why just remove on-predicates here.. Some cases `onPredicate ` are not kept in Join which can be pushed down to children. How about this?

Reply:
the core logical is that do not add repeat join on predicate into compensation query predicates. just consider the not-pushdown on predicates. Pushdown onPredicates is not necessary.